src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -403,6 +405,124 @@ public async Task CreateAsync(ContentItem contentItem, VersionOptions options)
             }
         }
 
+        public async Task<ContentValidateResult> ValidateAsync(ContentItem contentItem)
+        {
+            var context = new ValidateContentContext(contentItem);
+
+            await Handlers.InvokeAsync((handler, context) => handler.ValidateAsync(context), context, _logger);
+
+            if (!context.ContentValidateResult.Succeeded)
+            {
+                _session.Cancel();
+            }
+
+            return context.ContentValidateResult;
+        }
+
+        public Task<ContentValidateResult> CreateContentItemVersionAsync(ContentItem contentItem)
+        {
+            return CreateContentItemVersionAsync(contentItem, null);
+        }
+
+        public Task<ContentValidateResult> UpdateContentItemVersionAsync(ContentItem updatingVersion, ContentItem updatedVersion)
+        {
+            return UpdateContentItemVersionAsync(updatedVersion, updatedVersion);
+        }
+
+        public async Task ImportAsync(IEnumerable<ContentItem> contentItems)
+        {
+            var skip = 0;
+            var take = _importBatchSize;
+
+            var batchedContentItems = contentItems.Skip(skip).Take(take);
+
+            while (batchedContentItems.Any())
```

Comment:
Not sure but just because i thought about it. Maybe good to create a shell scope per batch. Unless we want a complete successful import or nothing